### PR TITLE
feat: improve deployment workflow

### PR DESCRIPTION
**Pull request by G3root**

### CODE REVIEW

File: .github/workflows/deploy-production.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Fly Production Deploy
+env:
+  FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
+on:
+  #   release:
+  #     types: [published]
```
Comment (G3root) [inline .github/workflows/deploy-production.yml:6]: should run when there is a release

File: .github/workflows/deploy-staging.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Fly Staging Deploy
+env:
+  FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
+on:
+  push:
+    # branches:
+    #   - main
```
Comment (G3root) [inline .github/workflows/deploy-staging.yml:7]: should run on every merge to the main

File: .github/workflows/deploy-staging.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+      # move Dockerfile to root
+      - name: 🚚 Move Dockerfile
+        run: |
+          mv ./docker/Dockerfile ./Dockerfile
+
+      - name: 🚀 Deploy Staging
+        run:
+          flyctl deploy --remote-only --build-arg COMMIT_SHA=${{ github.sha }}
+          --app ${{ steps.app_name.outputs.value }}-staging
```
Comment (G3root) [inline .github/workflows/deploy-staging.yml:35]: deploys to `${app_name}-staging`, where we can have separate env for staging

File: fly.toml [hunk 4]
```diff
@@ -6,25 +6,27 @@
...
-  cpu_kind = 'shared'
-  cpus = 1
+memory = '1gb'
+cpu_kind = 'shared'
+cpus = 1
 
 [deploy]
-  strategy = "canary"
+strategy = "canary"
+release_command = "npx prisma migrate deploy"
```
Comment (G3root) [inline fly.toml:29]: sync migration to production

File: fly.toml [hunk 5]
```diff
@@ -6,25 +6,27 @@
 app = 'captable'
 primary_region = 'lax'
 
-[build]
```
Comment (G3root) [inline fly.toml:9]: we don't need this because we copy the docker file to the root in github action

**Review by dahal — APPROVED** [review]

**Pull request merged**